### ISSUE TITLE
TabView throw a 'System.ArgumentException' while removing an item from it's TabItemsSource

### ISSUE THREAD

**Issue by Guillermo-Santos** [microsoft/microsoft-ui-xaml#7608]
### Describe the bug

I created a WinUI3 app using the template selector and added a tabview to work with files in the main page, i did this by making the tabitemsource a ObservableCollection of <Document>, the thing is that if i navigate to other page and then return to the main page, if i decide to open two or more tabs and then close one of the unselected tabs, the app 
`
Exception thrown: 'System.ArgumentException' in WinRT.Runtime.dll
`
`
WinRT information: The parameter is incorrect.
`

I think this is a bug, since i can open and close as many tabs as i want when i am on the main page (selected tabs or not) without issue, but if i navigate to other page an then return, closing new opened tabs or changing the selected tab and then closing it, throw this error if unselected.

### Steps to reproduce the bug

1. Open app.
2. Open a directory.
3. Click settings icon to go to settings page.
4. Click settings icon to go back to main page.
5. Open some documents
6. Close unselected document.

### Expected behavior

The document should close successfully but for some reason it does not.

### Screenshots

1. Open app.
2. Open a directory.
![Screenshot 2022-08-15 190135](https://user-images.githubusercontent.com/96314559/184735347-6c9443bf-df2a-4afb-98b4-27d9feb2e7d7.jpg)
3. Click settings icon to go to settings page.
![Screenshot 2022-08-15 191137](https://user-images.githubusercontent.com/96314559/184736055-dbe21e04-7438-475a-ba78-9b1c7e3d212e.jpg)
4. Click settings icon to go back to main page.
![Screenshot (52)](https://user-images.githubusercontent.com/96314559/184735632-8c5a71c0-a956-4ea5-9d0c-fdd00acfb977.png)
5. Open some documents
![Screenshot (55)](https://user-images.githubusercontent.com/96314559/184735690-4ccd3b62-b6b9-46bc-b32c-b1317d53320a.png)
6. Close unselected document.
![Screenshot 2022-08-15 183135](https://user-images.githubusercontent.com/96314559/184735488-cf561de4-4503-46e5-9eb7-0dc300a1f3fa.jpg)

### NuGet package version

WinUI 3 - Windows App SDK 1.1.4

### Windows app type

- [x] UWP
- [ ] Win32

### Device form factor

Desktop

### Windows version

Windows 11 (22H2): Build 22000

### TabView Control
```xml
<TabView x:Name="codeTabs"
         IsAddTabButtonVisible="False"
         CloseButtonOverlayMode="OnPointerOver"
         TabCloseRequested="TabView_TabCloseRequested"
         HorizontalAlignment="Stretch"
         VerticalAlignment="Stretch"
         TabItemsSource="{x:Bind ViewModel.Files}"
         SelectedItem="{x:Bind ViewModel.SelectedDocument, Mode=TwoWay}">
    <TabView.TabItemTemplate>
        <DataTemplate x:DataType="models:Document">
            <TabViewItem Header="{x:Bind FileName}">
                <TabViewItem.Content>
                    <local:FilePage FileName="{x:Bind Path}" Text="{x:Bind Text}"/>
                </TabViewItem.Content>
            </TabViewItem>
        </DataTemplate>
    </TabView.TabItemTemplate>
</TabView>
```
### TabView_TabCloseRequested
```cs    
    private void TabView_TabCloseRequested(TabView sender, TabViewTabCloseRequestedEventArgs args)
    {
        SparkFileService.Instance.CloseFile(args.Item as Document);
    }
```
### CloseFile
```cs
    public void CloseFile(Document document)
    {
        var openDocument = OpenDocuments.Where(d => d.Path == document.Path).FirstOrDefault();
        if (openDocument != null)
        {
            OpenDocuments.Remove(openDocument);
        }
    }
```
### Fix
```cs
    private void TabView_TabCloseRequested(TabView sender, TabViewTabCloseRequestedEventArgs args)
    {
        var document = args.Item as Document;
        var selectedDocument = ViewModel.SelectedDocument;
        ViewModel.SelectedDocument = document;
        SparkFileService.Instance.CloseFile(document);
        ViewModel.SelectedDocument = selectedDocument;
    }
```
This is a fix to the problem, since this bug only reproduce with unselected tabs, the solution i found was making the tab to close the selected tab and then closing it.

**Comment by enjoysmath** [comment]
I'm getting the same bug when I toggle my light / dark theme with the same old switch I've been using.  Now it suddenly crashes the app....  I can't make an MWE to reproduce it.

**Comment by ranjeshj** [comment]
@Guillermo-Santos can you please share a standalone repro app or the callstack/crashdump?

**Comment by Guillermo-Santos** [comment]
> @Guillermo-Santos can you please share a standalone repro app or the callstack/crashdump?

Of course, this is the repos [Guillermo-Santos/TheCompiler/tree/master/Compilador](https://github.com/Guillermo-Santos/TheCompiler/tree/master/Compilador) I cannot make modifications since the repo is archived but I will tell you the three things you will need to do:

1. You will need to mark the `Forge` project selfcontained (`<SelfContained>true</SelfContained>`).
2. You will need to undo the change i made on the `TabView_TabCloseRequested` method at the `TheCompiler/Compilador/Forge/Views/MainPage.xaml.cs` file. The code of that method is a temporal fix to the problem, it should be changed to the original I show above.
3. Once you build and run the app then use the app's option `File>Open Project` and go to `TheCompiler/Compilador/Samples/Default` and pick it's project file, then follow the steps to reproduce the bug.
